### PR TITLE
[DAT-3406] Indexer undertaker fixes

### PR DESCRIPTION
**Pull request by rybmat**
https://wikia-inc.atlassian.net/browse/DAT-3406

### CODE REVIEW

File: workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java [hunk 1]
```diff
@@ -27,22 +29,24 @@ public boolean handle(String message, QueueingConsumer.Delivery delivery) {
...
    * @return JSONObject
    */
-  public JSONObject headersToJSON(String headers) {
-    String jsonArray = headers.replaceAll("=", ":")
-        .replaceAll(":,", ":null,")     // sometimes there are keys without values
-        .replaceAll("time:[^,]*,", "")  // time is hard to parse
-        .replaceAll(":", ":\"")         // treat all values as strings
+  public JSONArray headersToJSON(String headers) {
+    String jsonArray = headers
+        .replaceAll("=", ":\"")         // treat all values as strings
```
Comment (WikiaTech) [inline workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java]: MINOR Move this trailing comment on the previous empty line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java [hunk 2]
```diff
@@ -27,22 +29,24 @@ public boolean handle(String message, QueueingConsumer.Delivery delivery) {
...
         .replaceAll(",", "\",")         // treat all values as strings
```
Comment (WikiaTech) [inline workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java]: MINOR Move this trailing comment on the previous empty line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java [hunk 3]
```diff
@@ -27,22 +29,24 @@ public boolean handle(String message, QueueingConsumer.Delivery delivery) {
...
-        .replaceAll("}", "\"}");        // closing "
-
-    return new JSONObject(jsonArray.substring(1, jsonArray.length() - 1)); // removing first [ and last ]
+        .replaceAll("=,", ":null,")     // sometimes there are keys without values
```
Comment (WikiaTech) [inline workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java]: MINOR Move this trailing comment on the previous empty line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java [hunk 4]
```diff
@@ -27,22 +29,24 @@ public boolean handle(String message, QueueingConsumer.Delivery delivery) {
...
+        .replaceAll("=", ":")           // key/value delimiter
```
Comment (WikiaTech) [inline workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java]: MINOR Move this trailing comment on the previous empty line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java [hunk 5]
```diff
@@ -27,22 +29,24 @@ public boolean handle(String message, QueueingConsumer.Delivery delivery) {
...
+        .replaceAll("}", "\"}")        // closing "
```
Comment (WikiaTech) [inline workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java]: MINOR Move this trailing comment on the previous empty line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java [hunk 6]
```diff
@@ -27,22 +29,24 @@ public boolean handle(String message, QueueingConsumer.Delivery delivery) {
...
+        .replaceAll("}\"", "}");       //fixing }" case
```
Comment (WikiaTech) [inline workers/indexer-undertaker/src/main/java/com/wikia/workers/indexerundertaker/IndexerUndertaker.java]: MINOR Move this trailing comment on the previous empty line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ATrailingCommentCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues:
* MINOR 6 minor

Watch the comments in this conversation to review them.

**Comment by SebastianMarzjan** [comment]
Please add a test that covers the  case reported in the ticket

**Comment by SebastianMarzjan** [comment]
LGTM
